tools/lib/provision_inner.py

```diff
@@ -93,8 +93,27 @@ def write_command(command: str) -> None:
             with open(shell_profile_path, "w") as shell_profile_file:
                 shell_profile_file.writelines(command + "\n")
 
-    source_activate_command = "source " + os.path.join(VENV_PATH, "bin", "activate")
-    write_command(source_activate_command)
+    def check_environment() -> str:
+        env_info = os.uname()
+        if "WSL" in env_info.release:
+            return "wsl"
+        elif os.getenv("VAGRANT_HOME"):
+            return "vagrant"
+        elif os.getenv("DOCKER_VERSION"):
+            return "docker"
+        elif os.path.exists("/etc/digitalocean"):
+            return "droplet"
+        else:
+            return "native linux"
```

Comment:
Let's move this to a top-level function. I don't like the WSL check -- `WSL` could be somewhere in `release` because `COWSLOT` is in there.

Reply:
I will move this to a top-level function. And will refactor the check for WSL from checking the release to file path. Specifically all WSL containers has `/proc/sys/fs/binfmt_misc/WSLInterop`.  So I think I'm gonna use that here.